flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -177,11 +179,19 @@ public IncrementalRocksDBSnapshotResources syncPrepareResources(long checkpointI
             return registry -> SnapshotResult.empty();
         }
 
+        final Set<StateHandleID> baseSstFiles;
+        if (checkpointOptions.getCheckpointType().getSharingFilesStrategy()
+                == CheckpointType.SharingFilesStrategy.FORWARD_BACKWARD) {
+            baseSstFiles = snapshotResources.baseSstFiles;
+        } else {
+            baseSstFiles = Collections.emptySet();
+        }
```

Comment:
Isn't this exactly the non future proof way of implementing this flag that I was mentioning?

shouldn't this be
```
switch (getSharingFilesStrategy()) ) {
  case FORWARD_BACKWARD:
     (...)
  case FORWARD:
      (...)
  case NO_SHARING:
  default:
     throw new UnsupportedOperationException();
}
```
?

Reply:
Aah ok, I get it now. Ok, I'll change it to switch.